http-filter-example/http_filter.cc

```diff
@@ -1,18 +1,60 @@
 #include <string>
+#include <unordered_map>
+#include <sstream>
+#include <vector>
 
 #include "http_filter.h"
 
+#include "source/common/common/logger.h"
 #include "envoy/server/filter_config.h"
 
 namespace Envoy {
 namespace Http {
 
 HttpSampleDecoderFilterConfig::HttpSampleDecoderFilterConfig(
     const sample::Decoder& proto_config)
-    : key_(proto_config.key()), val_(proto_config.val()) {}
+    : key_(proto_config.key()), val_(proto_config.val()), extra_(stoi(proto_config.extra())) {}
 
 HttpSampleDecoderFilter::HttpSampleDecoderFilter(HttpSampleDecoderFilterConfigSharedPtr config)
-    : config_(config) {}
+    : config_(config) {
+  // populate list of valid operations, implementation will be changed in the next few commits
+  operations_.insert("set-header");
+  
+  const std::string header_config = headerValue();
+
+  // find the position of the first space character
+  const size_t spacePos = header_config.find(' ');
+
+  const auto operation = header_config.substr(0, spacePos);
+
+  // error checking -- invalid operation
+  auto it = operations_.find(operation);
+  if (it == operations_.end()) {
+    ENVOY_LOG_MISC(info, "invalid config given -- invalid operation");
+    setError(1);
```

Comment:
Should this immediately `return`? Same question for the `setError` below

Reply:
I was thinking it would be good to use different numbers, and define them as constants, so we can distinguish different errors.

But I like this pattern even better:
https://github.com/envoyproxy/envoy/blob/c49f3925/source/extensions/tracers/xray/localized_sampling.cc#L28C1-L32C2
```
namespace {
void fail(absl::string_view msg) {
  auto& logger = Logger::Registry::getLog(Logger::Id::tracing);
  ENVOY_LOG_TO_LOGGER(logger, error, "Failed to parse sampling rules - {}", msg);
}
```
```
    fail("missing version number");
    return;
```